<launch>
    <arg name="visualize" default="true"/>
    <arg name="odom_frame" default="odom"/>
    <arg name="base_frame" default="base_link"/>
    <arg name="topic" default=""/>

    <!-- set config parameters -->
    <arg name="deskew" default="false"/>
    <arg name="max_range" default="100.0" />
    <arg name="min_range" default="5.0" />
    <arg name="voxel_size" default="0.5" />

    <arg name="pose_extractor" default="false"/>
    <arg name="rviz" default="true"/>
    <param name="use_sim_time" value="true"/>
    <node pkg="lio_sphere" type="lio_node" name="lio_node" output="screen" if="true">
        <!-- Remap Topics -->
        <remap from="/points_in" to="/hesai/pandar"/>
        <remap from="/imu_in" to="/posePub_merged"/>
        <remap from="/lio/preprocessed" to="/lio/preprocessed"/>
        <param name="use_sim_time" value="true"/>

        <param name="odom_frame" value="$(arg odom_frame)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="visualize" value="$(arg visualize)"/>
        <!-- Config Params -->
        <param name="max_range" value="$(arg max_range)"/>
        <param name="min_range" value="$(arg min_range)"/>
        <param name="deskew" value="$(arg deskew)"/>
        <param name="voxel_size" value="$(arg voxel_size)"/>
        <param name="max_points_per_voxel" value="20"/>
        <param name="initial_threshold" value="2.0"/>
    </node>
</launch>